### ISSUE TITLE
In-repo-addon within an addon fails

### ISSUE THREAD

**Issue by aceofspades** [ember-cli/ember-cli#4918]
It may be useful to group a number of shared in-repo-addons within another wrapper addon. This seems to fail when building the app:

Sample: https://github.com/aceofspades/ember-cli-addon-with-ira

```
version: 1.13.8
Could not find watchman, falling back to NodeWatcher for file system events.
Visit http://www.ember-cli.com/user-guide/#watchman for more info.
Cannot read property 'addonLintTree' of undefined
TypeError: Cannot read property 'addonLintTree' of undefined
    at Class.Addon.jshintAddonTree (/private/tmp/tmp/myapp/node_modules/ember-cli/lib/models/addon.js:543:31)
    at Class.treeFor (/private/tmp/tmp/myapp/node_modules/ember-cli/lib/models/addon.js:305:21)
    at /private/tmp/tmp/myapp/node_modules/ember-cli/lib/models/addon.js:253:32
    at Array.map (native)
    at Class.eachAddonInvoke (/private/tmp/tmp/myapp/node_modules/ember-cli/lib/models/addon.js:251:22)
    at Class.treeFor (/private/tmp/tmp/myapp/node_modules/ember-cli/lib/models/addon.js:298:20)
    at EmberApp.<anonymous> (/private/tmp/tmp/myapp/node_modules/ember-cli/lib/broccoli/ember-app.js:358:20)
    at Array.map (native)
    at EmberApp.addonTreesFor (/private/tmp/tmp/myapp/node_modules/ember-cli/lib/broccoli/ember-app.js:356:30)
    at EmberApp._processedAppTree (/private/tmp/tmp/myapp/node_modules/ember-cli/lib/broccoli/ember-app.js:616:25)
```

**Comment by stefanpenner** [comment]
this is not currently supported.

**Comment by runspired** [comment]
This error also occurs when including an addon in another addons npm `dependencies`, breaking addons that necessitate other addons

**Comment by trentmwillis** [comment]
I've been running into this frequently while trying to utilize Engines. [A workaround exists by turning off `hintingEnabled`](https://github.com/dgeb/ember-engines/blob/master/tests/dummy/lib/ember-blog/index.js#L11-L13), but this is obviously less than ideal.

I'll take a look into it and see if I can find an actual solution.

**Comment by ro0gr** [comment]
It's great to see it fixed!   
Just curious about is it supposed to be merged to 1.13?

**Comment by Turbo87** [comment]
apparently this got fixed in #5432
